tools/clang/test/CodeGenHLSL/quick-test/empty_struct3.hlsl

```diff
@@ -0,0 +1,28 @@
+// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s
+
+// Make sure nest empty struct works.
```

Comment:
"Regression test for a crash where empty structs something something with calls"

Also, if you need nested classes and cbuffers containing arrays and unused `pos` input, explain why, otherwise, simplify the test.

Reply:
This tests a number of code paths that can mishandle non-trivially empty aggregates, simultaneously.  I will try to elaborate in comments in the test.  `pos` can be removed, of course.